packages/types/src/DidDetails.ts

```diff
@@ -218,3 +164,15 @@ export type DidSignature = {
 export type TypedValue<K extends string, V> = K extends any
   ? { [P in K]: V }
   : never
+
+export interface DidDetails {
+  identifier: DidIdentifier
+  uri: DidUri
+
+  authentication: [DidVerificationKey]
+  assertionMethod?: [DidVerificationKey]
+  capabilityDelegation?: [DidVerificationKey]
+  keyAgreement?: DidEncryptionKey[]
+
+  service?: DidServiceEndpoint[]
```

Comment:
Technically, the ID of each service endpoint is unique in regard to other endpoints but also in regard with the keys. I am not fully happy with the fact that we model them as an array. Of course the data comes from the chain, where it is assumed to be always correct, but it would make more sense to model the services as a record (or map), and the same for the keys, since it would then provide more efficient lookups.

Reply:
The idea is to literally work with the DID document. And the DID Document has them as an array. The performance impact on having to iterate over the array, every time you want to access a specific service endpoint is negligible.

Reply:
The spec defines the property as a SET, meaning that each element (identified by its ID) cannot appear more than once -> https://www.w3.org/TR/did-core/#services. Just saying that we could enforce this at compilation time since at least with Typescript.

Reply:
I feel that enforcing unique IDs across all DID properties is a complex task and that TypeScript is a wrong tool to solve it. Not only would this solution be incomplete, it will result in awkward data structures. With the lean SDK being a goal, I would avoid even attempting to solve the issue in it.

Reply:
This depends on whether we want to go all in on the idea "literally work with the DID document" or just move in this direction. So far I believe that some practical differences are warranted: 

1) use key type `sr25519` instead of `Sr25519VerificationKey2020`
2) when the same key was used for multiple relationships, duplicate it instead of referencing an entry in `verificationMethods` by ID
3) consider the fragment part as "internal ID" and use it

The transformation to a proper interchangeable DID Document could be done by the DocumentExporter

Reply:
yeah these are good points, switching to referencing key types as `Sr25519VerificationKey2020` sounds like asking for trouble. And #2 is just a consequence of the simplified data model of the did document that we adopt in this PR. I can't really say what the most practical position is on the question just how much we want to go 'all-in' on the "literally work with the DID document" goal.